File: src/main/java/com/braggbnb102/service/ResponseService.java
package com.braggbnb102.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.braggbnb102.domain.Response;
import com.braggbnb102.dto.ResponseDTO;
import com.braggbnb102.dto.ResponseSearchDTO;
import com.braggbnb102.dto.ResponsePageDTO;
import com.braggbnb102.dto.ResponseConvertCriteriaDTO;
import com.braggbnb102.service.GenericService;
import com.braggbnb102.dto.common.RequestDTO;
import com.braggbnb102.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ResponseService extends GenericService<Response, Integer> {

	List<Response> findAll();

	ResultDTO addResponse(ResponseDTO responseDTO, RequestDTO requestDTO);

	ResultDTO updateResponse(ResponseDTO responseDTO, RequestDTO requestDTO);

    Page<Response> getAllResponses(Pageable pageable);

    Page<Response> getAllResponses(Specification<Response> spec, Pageable pageable);

	ResponseEntity<ResponsePageDTO> getResponses(ResponseSearchDTO responseSearchDTO);
	
	List<ResponseDTO> convertResponsesToResponseDTOs(List<Response> responses, ResponseConvertCriteriaDTO convertCriteria);

	ResponseDTO getResponseDTOById(Integer responseId);







}

File: braggbnb102_ui/src/services/ResponseService.js
import http from "../http-common"; 

class ResponseService {
  getAllResponses(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/response/responses`, searchDTO);
  }

  get(responseId) {
    return this.getRequest(`/response/${responseId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/response?field=${matchData}`, null);
  }

  addResponse(data) {
    return http.post("/response/addResponse", data);
  }

  update(data) {
  	return http.post("/response/updateResponse", data);
  }
  
  uploadImage(data,responseId) {
  	return http.postForm("/response/uploadImage/"+responseId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ResponseService();
